composer.json

```diff
@@ -25,6 +25,7 @@
         "ext-json": "*",
         "cebe/php-openapi": "^1.3",
         "dflydev/fig-cookies": "^2.0",
+        "league/uri": "^6.3",
```

Comment:
I suppose the main problem with this approach is that it could cause BC issues for a user who is already dependent on a different version of `league/uri`.

Reply:
Being dependend on `^6.1` for example is not a BC when requesting a lib bringing requirement of `^6.3`. It'will just make you to update to the `^6.3` (which comforms `^6.1`). The problems are then constraints are misused, like `~6.2.1` or `6.0.3` or whatever unnecessary strict. 

This should be resolved by the people who decided to make these constraints.

If people really struggle to bring `^6.3` to their project - downgrading dependency to `^6.x` is good until it's working as same as `^6.3` in terms of our usage. And such a downgrade is backward compatible

Reply:
What about the case where someone already has a dependency on `5.1` and doesn't want to, or can't, upgrade to 6.x? When they try to upgrade the openapi validator package, composer will error because there will be a dependency conflict between 6.x (this package) and 5.x (their other dependency).

Reply:
Supporting out of date dependencies is always the matter of additional resources. Why we are talking only about 5.x, not 4.x, 3.x, 2.x or people who's dependensies do not allow to bring the lib at all (imagine they have another dependency, which have `conflicts: {"league/uri":"*"}`)

In general we do support only maintained verions of dependecies (latest minor by default, according to semver). Any wider assumptions MAY be implemented, but with real use cases and better with resources from people who really need this

Reply:
Sorry, maybe I wasn't explaining myself well :)

I am not suggesting that you should support out-of-date dependencies. If we want to use `league/uri` as a dependency, we absolutely should use the latest version. I just wanted to flag up that if there is someone who depends on `league/uri` in any previous version before the `6.x` line, when they come to update *this* library, they will get a composer error, and will be forced to update `league/uri` as well.

Anyway, do you want me to leave this as dependent on `^6.3` or change to something else? I'm happy to do whatever you'd prefer. I will also fix the PHPCS errors from the Travis build.

Reply:
> when they come to update this library, they will get a composer error, and will be forced to update league/uri as well.

or they just stick to version of this lib they already have. no BC-breaks here.

> Anyway, do you want me to leave this as dependent on ^6.3 or change to something else? I'm happy to do whatever you'd prefer

`^6.3` is fine at the moment. We can downgrade the dependecy at any further moment easy.